packages/react/src/compatible-components/radio-group/radio-group.tsx

```diff
@@ -0,0 +1,56 @@
+import { RadioButton } from '../../components/radio-button';
+import {
+  RadioGroup,
+  RadioGroupProps,
+  RadioGroupValue,
+} from '../../components/radio-group';
+
+// Types temporary taken from devextreme ------
+interface CollectionWidgetItem {
+  disabled?: boolean;
+  html?: string;
+  // template?: ....
+  text?: string;
+  visible?: boolean;
+}
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type ItemLike = string | CollectionWidgetItem | any;
+//------------------------------
+
+const createItemPropGetter = (propName: string) => (item: ItemLike) => (
+  Object.prototype.hasOwnProperty.call(item, propName) ? item[propName] : item
+);
+
+interface CompatibleRadioGroupProps<T> extends RadioGroupProps<T> {
+  items: Array<ItemLike>;
+  // eslint-disable-next-line react/require-default-props
+  itemRender?: (data: ItemLike, index?: number) => JSX.Element;
+  // eslint-disable-next-line react/require-default-props
+  valueExpr?: string;
+  // eslint-disable-next-line react/require-default-props
+  displayExpr?: string;
+}
+
+const valuePropNameDefault = 'text';
+
+export function RadioGroupCompatible({
+  items,
+  itemRender,
+  defaultValue,
+  valueExpr,
+  displayExpr,
+}: CompatibleRadioGroupProps<RadioGroupValue>) {
+  const getItemLabel = createItemPropGetter(displayExpr || valuePropNameDefault);
+  const getItemValue = createItemPropGetter(valueExpr || valuePropNameDefault);
+
+  return (
+    <RadioGroup defaultValue={defaultValue}>
+      {items.map((item, index) => (
+        <RadioButton
+          value={getItemValue(item)}
+          label={itemRender ? itemRender(item, index) : getItemLabel(item)}
```

Comment:
Move this logic into the radio button component and leave following:
```
<RadioButton
...
label={getItemLabel(item)}
labelTemplate={itemRender}
labelComponnet={itemComponent}
/>
```

Reply:
The types of renovated RadioButton and DevExtreme RadioGroup props are incompatible.
Old RadioGroup's labelComponent accepts { data } as props, where data is the item itself (of ItemLike type). 
RadioButton's labelTemplate accepts the label contents passed to RadioButton's label prop.
I think it is better to encapsulate all compatibility logic in the RadioGroupCompatible component.